src/Microsoft.OpenApi/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OpenApi.Extensions
+{
+    /// <summary>
+    /// Dictionary extension methods.
+    /// </summary>
+    public static class DictionaryExtensions
+    {
+        /// <summary>
+        /// Returns a new dictionary with entries sorted by key using the default comparer.
+        /// </summary>
+        public static IDictionary<TKey, TValue> Sort<TKey, TValue>(
+            this IDictionary<TKey, TValue> source)
+            where TKey : notnull
+        {
+            if (source == null)
```

Comment:
we have util methods in the library for null checks, please use those for consistency